src/integrationTest/java/uk/gov/hmcts/reform/translate/repository/DictionaryRepositoryIT.java

```diff
@@ -0,0 +1,107 @@
+
+import org.assertj.core.util.Streams;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.context.jdbc.Sql;
+import uk.gov.hmcts.reform.translate.data.DictionaryEntity;
+import uk.gov.hmcts.reform.translate.data.TranslationUploadEntity;
+
+import java.time.LocalDateTime;
+import java.util.UUID;
+
+import static org.junit.jupiter.api.Assertions.assertAll;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+@SpringBootTest
+@SuppressWarnings("PMD.JUnitAssertionsShouldIncludeMessage")
+@ActiveProfiles("itest")
+class DictionaryRepositoryIT {
+    @Autowired
+    DictionaryRepository dictionaryRepository;
+
+    private static final String ENGLISH_PHRASE = "English phrase";
+    private static final String TRANSLATION_PHRASE = "Welsh translation phrase";
+    private static final String IDAM_USER_ID = UUID.randomUUID().toString();
+
+    private static final String GET_TRANSLATION_TABLES_SCRIPT =
+        "classpath:sql/get-Dictionary_And_TranslationUploads.sql";
+    private static final String DELETE_TRANSLATION_TABLES_SCRIPT =
+        "classpath:sql/delete-Dictionary_And_TranslationUploads.sql";
+
+    @Sql(scripts = DELETE_TRANSLATION_TABLES_SCRIPT)
+    @Test
+    void testSaveDictionaryAndTranslationUpload() {
+        final var now = LocalDateTime.now();
+        final var translationUploadEntity = new TranslationUploadEntity();
+        translationUploadEntity.setUploaded(now);
+        translationUploadEntity.setUserId(IDAM_USER_ID);
+
+        final var dictionaryEntity = new DictionaryEntity();
+        dictionaryEntity.setEnglishPhrase(ENGLISH_PHRASE);
+        dictionaryEntity.setTranslationPhrase(TRANSLATION_PHRASE);
+        dictionaryEntity.setTranslationUpload(translationUploadEntity);
+
+        final var dictionary = dictionaryRepository.save(dictionaryEntity);
+
+        assertNotNull(dictionary.getId());
+
+        assertAll(
+            () -> assertEquals(TRANSLATION_PHRASE, dictionary.getTranslationPhrase()),
+            () -> assertEquals(ENGLISH_PHRASE, dictionary.getEnglishPhrase()),
+            () -> assertEquals(now, dictionary.getTranslationUpload().getUploaded()),
+            () -> assertEquals(IDAM_USER_ID, dictionary.getTranslationUpload().getUserId())
+        );
+    }
+
+    @Sql(scripts = {DELETE_TRANSLATION_TABLES_SCRIPT, GET_TRANSLATION_TABLES_SCRIPT})
+    @Test
+    void testFindDictionaryAndTranslationUploadNoTranslationPhrase() {
+        final var optionalDictionaryEntity = dictionaryRepository.findById(1);
+        assertTrue(optionalDictionaryEntity.isPresent());
```

Comment:
you could use val, instead of final var

Reply:
'val' is a lombok thing - while I like lombok for annotating classes, I'm not so keen on using it here...will leave as is for  now